src/PhpSpec/CodeAnalysis/AccessInspector.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace PhpSpec\CodeAnalysis;
+
+use ReflectionMethod;
+use ReflectionProperty;
+
+class AccessInspector implements AccessInspectorInterface
```

Comment:
It's hard to answer what makes this specific without reflecting on the intention behind the PR, which I'm trying not to do. maybe DetailedAccessInspector, MagicAccessInspector, GenericAccessInspector?